packages/malloy/src/lang/malloy-to-ast.ts

```diff
@@ -382,7 +419,13 @@ export class MalloyToAST
   ): ast.HasParameter[] {
     if (pcx === undefined) return [];
     this.inExperiment('parameters', pcx);
-    return pcx.sourceParameter().map(param => this.getSourceParameter(param));
+    function notNullParam(p: ast.HasParameter | null): p is ast.HasParameter {
```

Comment:
Actually this is another thing I would like to talk to chris about

If there is an error in a parameter declaration, is it better to NOT define the parameter, or to somehow make it a "parameter with type 'error'"

Reply:
Probably better to make it a parameter with type error, so that if you have

```
source: x(foo is r'asdf') ...
run: x(foo is r'fdas') -> ...
```
You get "parameter cannot be regular expression" but not also "x has no parameter foo"

Reply:
The problem is that 'error' is not a legal type for creating a `HasParameter` and I didn't know if that was intentional for some reason.